rust/main/chains/hyperlane-ton/src/contracts/interchain_gas.rs

```diff
@@ -68,70 +67,9 @@ impl Indexer<InterchainGasPayment> for TonInterchainGasPaymasterIndexer {
     ) -> ChainResult<Vec<(Indexed<InterchainGasPayment>, LogMeta)>> {
         let start_block = max(*range.start(), 1);
         let end_block = max(*range.end(), 1);
-        let start_block_info = self
-            .provider
-            .get_blocks(
-                -1,                //  masterchain (workchain = -1)
-                None,              // shard
-                None,              // block seqno
-                Some(start_block), // masterchain seqno
-                None,
-                None,
-                None,
-                None,
-                None, // limit
-                None,
-                None,
-            )
-            .await
-            .map_err(|e| {
-                ChainCommunicationError::CustomError(format!(
-                    "Failed to get start block info: {:?}",
-                    e
-                ))
-            })?;
 
-        info!("Start block info:{:?}", start_block_info);
-        let end_block_info = self
-            .provider
-            .get_blocks(
-                -1,              //  masterchain (workchain = -1)
-                None,            // shard
-                None,            // block seqno
-                Some(end_block), // masterchain seqno
-                None,
-                None,
-                None,
-                None,
-                None,
-                None,
-                None,
-            )
-            .await
-            .map_err(|e| {
-                ChainCommunicationError::CustomError(format!(
-                    "Failed to get end block info: {:?}",
-                    e
-                ))
-            })?;
-
-        info!("End block info:{:?}", end_block_info);
-
-        let start_utime = start_block_info.blocks[0]
-            .gen_utime
-            .parse::<i64>()
-            .map_err(|e| {
-                ChainCommunicationError::CustomError(format!(
-                    "Failed to parse start_utime: {:?}",
-                    e
-                ))
-            })?;
-        let end_utime = end_block_info.blocks[0]
-            .gen_utime
-            .parse::<i64>()
-            .map_err(|e| {
-                ChainCommunicationError::CustomError(format!("Failed to parse end_utime: {:?}", e))
-            })?;
+        let start_utime = self.provider.fetch_block_timestamp(start_block).await?;
+        let end_utime = self.provider.fetch_block_timestamp(end_block).await?;
```

Comment:
I guess you can join this 2 call into 1 request - request two blocks with two secno